test/metabase/sync_database/analyze_test.clj

```diff
@@ -72,3 +78,80 @@
 (expect false (values-are-valid-emails? [100]))
 (expect false (values-are-valid-emails? ["true"]))
 (expect false (values-are-valid-emails? ["false"]))
+
+;; Tests to avoid analyzing hidden tables
+(defn count-un-analyzed-fields [db_id table-name]
+  (let [table-id (db/select-one-id table/Table, :db_id db_id, :active true :name table-name)]
+    (assert (pos? ;; don't let ourselves be fooled if the test passes because the table is
+                  ;; totally broken or has no fields. Make sure we actually test something
+             (db/count metabase.models.field/Field :table_id table-id)))
+    (db/count metabase.models.field/Field :last_analyzed nil :table_id table-id)))
+
+(defn get-latest-sync-time [db_id table-name]
+  (let [table-id (db/select-one-id table/Table, :db_id db_id, :active true :name table-name)]
+    (db/select-field :last_analyzed metabase.models.field/Field :table_id table-id)))
+
+
+(defn api-table-call [state table]
```

Comment:
`api-table-call` doesn't really do a good job of explaining what this function does

how about `set-table-visibility-type!` or something like that?